### PR TITLE
Moved to cadquery jupyter

### PR DESCRIPTION
**Pull request by shimwell**
## Proposed changes

This PR :

adds jupyter-cadquery to the docker image 
adds notebook versions of all the shape examples
adds new docker run commands to the readme and read the docs
adds tests for the new notebook files

The result is that we can now render shapes (not compounds) in the jupyter lab browser

Screenshot from 2021-03-03 15-44-46

## Types of changes

What types of changes does your code introduce to the Paramak?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [x] New tests

## Checklist

- [ ] Pep8 applied
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by shimwell** [comment]
This is failing as the dependencyimage does not have jupyter_cadquery installed. However jupyter_cadquery is being added by this PR. So I might break this into two PRs and first made a new docker image then add the new tests that use jupyter_cadquery

File: tests/test_example_shapes.py [hunk 1]
```diff
@@ -4,19 +4,57 @@
...
+                print(str(e.traceback).split("\n")[-2])
+            else:
+                raise e
+
+    return note_book, errors
+
+
 class TestExampleShapes(unittest.TestCase):
 
+    def test_task_1(self):
```
Comment (shimwell) [inline tests/test_example_shapes.py]:
```suggestion
    def cadquery_jupyter_test_task_1(self):
```

File: Dockerfile [hunk 2]
```diff
@@ -62,12 +62,12 @@ RUN apt-get install -y libgl1-mesa-glx libgl1-mesa-dev libglu1-mesa-dev \
                        apt-get clean
 
 # Installing CadQuery
-# jupyter is installed before cadquery to avoid a conflict
 RUN echo installing CadQuery version $cq_version && \
-    conda install jupyter -y --quiet && \
-    conda install -c cadquery -c conda-forge cadquery="$cq_version" && \
+    conda install -c cadquery -c conda-forge python=3.8 cadquery="$cq_version" && \
+    pip install jupyter-cadquery==2.0.0-rc1 && \
```
Comment (shimwell) [inline Dockerfile]:
```suggestion
    pip install jupyter-cadquery==2.0.0 && \
```

File: tests/test_example_shapes.py [hunk 3]
```diff
@@ -4,19 +4,57 @@
...
+
+
 class TestExampleShapes(unittest.TestCase):
 
+    def cadquery_jupyter_test_task_1(self):
+        for notebook in Path().rglob("examples/example_parametric_shapes/*.ipynb"):
+            print(notebook)
+            nb, errors = _notebook_run(notebook)
+            assert errors == []
+
```
Comment (shimwell) [inline tests/test_example_shapes.py]:
```suggestion
#    def cadquery_jupyter_test_task_1(self):
 #       for notebook in Path().rglob("examples/example_parametric_shapes/*.ipynb"):
#            print(notebook)
#            nb, errors = _notebook_run(notebook)
  #          assert errors == []

```

**Comment by shimwell** [comment]
Compounds are working thanks to feature addition in jupyter-cadquery 2.0.0

Screenshot from 2021-03-06 22-01-54

**Comment by shimwell** [comment]
I have now wrapped the jupyter-cadquery.cadquery show method and provided this as a Shape.show or Reactor.show attribute. I think this makes it easier for users and avoids one more import.

**Comment by shimwell** [comment]
I have added some new tests for the jupyter notebooks but these have not been enabled yet as the circle ci docker image doesn't have jupyter-cadquery till this PR is merged. So I can enable them in a subsequent PR.

**Review by billingsley-john — APPROVED** [review]
All looks great! These are nice additions

**Pull request merged**